include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -164,12 +226,58 @@ class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 		IDescriptorSetLayout(const SBinding* const _begin, const SBinding* const _end) : 
 			m_bindings((_end-_begin) ? core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<SBinding>>(_end-_begin) : nullptr)
 		{
+			struct SBindingRedirectBuildInfo
+			{
+				uint32_t binding;
+				uint32_t count;
+
+				inline bool operator< (const SBindingRedirectBuildInfo& other) const { return binding < other.binding; }
+			};
+
+			core::vector<SBindingRedirectBuildInfo> buildInfo_descriptors[asset::EDT_COUNT];
+			core::vector<SBindingRedirectBuildInfo> buildInfo_samplers;
+
+			for (auto b = _begin; b != _end; ++b)
+			{
+				buildInfo_descriptors[b->type].emplace_back(b->binding, b->count);
+				if (b->type == EDT_COMBINED_IMAGE_SAMPLER && b->samplers == nullptr)
+				{
+					buildInfo_samplers.emplace_back(b->binding, b->count);
+					m_mutableSamplerCount += b->count;
```

Comment:
`m_mutableSamplerCount` you can get from the inclusive scan later (lets switch away from exclusive because the first element is always 0 there)

Reply:
Resolved.